diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7125773..8836e3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ option(KDEPIM_MOBILE_UI "Build UI for mobile devices instead of for desktops" FA
 option(KDEPIM_ONLY_KLEO "Only build Kleopatra. This option will build only libkleo and kleopatra" FALSE)
 option(KDEPIM_BUILD_STATIC "Build KDEPIM static." FALSE)
 option(KDEPIM_BUILD_DESKTOP "Build Desktop Applications. Can be deactivated for mobile" TRUE)
-option(KDEPIM_NO_WEBKIT "Do not use WebKit in the kdepim applications" FALSE)
+option(KDEPIM_NO_WEBKIT "Do not use WebKit in the kdepim applications" TRUE)

 if(KDEPIM_BUILD_STATIC)
   set(LIBRARY_TYPE STATIC)
@@ -309,7 +309,6 @@ else()
     # move to runtime this can be avoided
     add_subdirectory(runtime)
   endif()
-  add_subdirectory(templateparser)

   if(QGPGME_FOUND)
     if(Boost_TOPOLOGICAL_SORT_DIR)
@@ -317,14 +316,10 @@ else()
     endif()
   endif()

-  # The following components depend on QGpgME.
-  set(MESSAGEVIEWER_SUPPORTED TRUE)
-  add_subdirectory(messageviewer)
-  macro_optional_add_subdirectory(messagecomposer)
   add_subdirectory(pimcommon)
-  add_subdirectory(mailcommon) # TODO: does this make sense?!?
-  add_subdirectory(mailfilteragent)
-  macro_optional_add_subdirectory(kmail)
+
+  # The following components depend on QGpgME.
+  set(MESSAGEVIEWER_SUPPORTED FALSE)

   if(KDEPIM_BUILD_MOBILE)
     add_subdirectory(mobile)
@@ -334,50 +329,24 @@ else()
     add_subdirectory(examples)
   endif()

-  # If kmail is compiled, KMAIL_SUPPORTED is true (used in several places)
-  if(BUILD_kmail)
-    set(KMAIL_SUPPORTED TRUE)
-    add_definitions(-DKMAIL_SUPPORTED)
-  endif()
+  set(BUILD_kmail FALSE)
+  set(KMAIL_SUPPORTED FALSE)
+  set(BUILD_knode FALSE)

   macro_optional_add_subdirectory(korganizer)
   if(KDEPIM_BUILD_DESKTOP)
     macro_optional_add_subdirectory(akregator)
-    macro_optional_add_subdirectory(archivemailagent)
-    macro_optional_add_subdirectory(importwizard)
     macro_optional_add_subdirectory(kaddressbook)
-    macro_optional_add_subdirectory(kmailcvt)
     macro_optional_add_subdirectory(knotes)
-    macro_optional_add_subdirectory(ksendemail)
-    macro_optional_add_subdirectory(ktnef)
-    macro_optional_add_subdirectory(mailimporter)
-    macro_optional_add_subdirectory(pimsettingexporter)
     macro_optional_add_subdirectory(kalarm)
-
-    if(KDEPIMLIBS_KCAL_LIBRARY)
-      macro_optional_add_subdirectory(blogilo)
-    endif()
-
-    if(Grantlee_FOUND)
-      macro_optional_add_subdirectory(kjots)
-    endif()
-
     if(KDEPIMLIBS_KRESOURCES_LIBRARY)
-      if(QT_QT3SUPPORT_FOUND)
-        macro_optional_add_subdirectory(knode)
-      endif()
-
       if(Q_WS_X11)
         macro_optional_add_subdirectory(ktimetracker)
       endif()
-
     endif()
-    macro_optional_add_subdirectory(kontact) # must be the last one.
   endif()

-  macro_optional_add_subdirectory(akonadiconsole)
   macro_optional_add_subdirectory(console)
-  macro_optional_add_subdirectory(strigi-analyzer)

   # These targets depend on optional applications
   if(KDEPIMLIBS_KRESOURCES_LIBRARY)
