diff -up kdepim-4.10.2/akregator/plugins/mk4storage/CMakeLists.txt.me kdepim-4.10.2/akregator/plugins/mk4storage/CMakeLists.txt
--- kdepim-4.10.2/akregator/plugins/mk4storage/CMakeLists.txt.me	2013-03-01 07:57:27.375648031 +0100
+++ kdepim-4.10.2/akregator/plugins/mk4storage/CMakeLists.txt	2013-04-29 14:41:15.224497773 +0200
@@ -1,3 +1,5 @@
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-aliasing")
+
 include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}
